<launch>
    <!-- Parameters -->
    <arg name="setup_file" default="euroc_mono_inertial_rudpt.launch" />
    <arg name="pose_topic" default="/orb_slam3/camera_pose" />
    <arg name="output_file" default="$(find rudpt_orb_slam3)/results/test.txt" />
    
    <!-- Setup the svo environment with the given setup file -->
    <include file="$(find rudpt_orb_slam3)/launch/$(arg setup_file)">
        <arg name="rviz_on" value="false" />
        <arg name="enable_pangolin" value="false" />
        <arg name="trajectory_server_on" value="true" />
    </include>
  
    <!-- Initialise node for saving the svo poses -->
    <node name="orb_slam3_to_txt" pkg="rudpt_orb_slam3" type="orb_slam3_pose_to_txt.py" output="screen" clear_params="true">
       <param name="pose_topic" value="$(arg pose_topic)" type="str" /> 
       <param name="output_file" value="$(arg output_file)" type="str" />
    </node>

</launch>
